File: captcha-frontend/captcha-verification/src/components/testing_methods/JS_testing_methods/botsimulator/behaviors/mousemove.js
const { randomInt, sleep } = require('../utils/random');

async function mouseMove(page) {
  const box = await page.viewport();
  const movements = randomInt(10, 50);
  for (let i = 0; i < movements; i++) {
    const x = randomInt(0, box.width);
    const y = randomInt(0, box.height);
    await page.mouse.move(x, y, { steps: randomInt(5, 25) });
    await sleep(randomInt(50, 200));
  }
}
module.exports = mouseMove;

File: captcha-frontend/captcha-verification/src/components/testing_methods/python testing methods/stealth_bot-1/behaviors/mouse_move.py
import random
import time
from selenium.webdriver.common.action_chains import ActionChains

def mouse_move(driver):
    print("🖱️ Simulating mouse movement...")

    actions = ActionChains(driver)
    
    # Get the window size to avoid going out of bounds
    window_size = driver.get_window_size()
    width = window_size['width']
    height = window_size['height']

    # Choose safe offsets (within visible window)
    x_offset = random.randint(10, width // 2)
    y_offset = random.randint(10, height // 2)

    try:
        actions.move_by_offset(x_offset, y_offset).perform()
        print(f"✅ Moved mouse by offset: ({x_offset}, {y_offset})")
    except Exception as e:
        print(f"⚠️ Failed to move mouse: {e}")
    
    time.sleep(random.uniform(0.5, 1.5))
